lib/galaxy/model/__init__.py

```diff
@@ -4146,6 +4148,37 @@ def add_input(self, content, step_id):
             request_to_content.workflow_step_id = step_id
             self.input_step_parameters.append(request_to_content)
 
+    # TODO: How is this caching supposed to work? When adding the _resource_parameters variable to the object, it does
+    # not get shown when accessing the object later and fails on 'if self._resource_parameters is None:'.
+
+    # @property
+    # def resource_parameters(self):
+    #     import pdb
+    #     pdb.set_trace()
+    #     if self._resource_parameters is None:
+    #         self.__cache_resource_parameters()
+    #     return self._resource_parameters
+    #
+    # def __cache_resource_parameters(self):
+    #     # TODO: prefetch and optimize down the road.
+    #     resource_type = WorkflowRequestInputParameter.types.RESOURCE_PARAMETERS
+    #     resource_parameters = {}
+    #     for input_parameter in self.input_parameters:
+    #         if input_parameter.type == resource_type:
+    #             resource_parameters[input_parameter.name] = input_parameter.value
+    #
+    #     self._resource_parameters = resource_parameters
+
+    # TODO: Remove this. Temp solution until caching is fixed.
```

Comment:
This is good for now. Drop the TODO and delete the cached stuff I think.

Reply:
Add a ``@property`` decorator to this method and drop the parens in the next TODO down. https://docs.python.org/2/library/functions.html#property